fs-api-blockio/src/System/FS/BlockIO/API.hs

```diff
@@ -60,5 +75,30 @@ ioopHandle :: IOOp m h -> Handle h
 ioopHandle (IOOpRead h _ _ _ _)  = h
 ioopHandle (IOOpWrite h _ _ _ _) = h
 
+ioopFileOffset :: IOOp m h -> FileOffset
+ioopFileOffset (IOOpRead _ off _ _ _)  = off
+ioopFileOffset (IOOpWrite _ off _ _ _) = off
+
+ioopBuffer :: IOOp m h -> MutableByteArray (PrimState m)
+ioopBuffer (IOOpRead _ _ buf _ _)  = buf
+ioopBuffer (IOOpWrite _ _ buf _ _) = buf
+
+ioopBufferOffset :: IOOp m h -> BufferOffset
+ioopBufferOffset (IOOpRead _ _ _ bufOff _)  = bufOff
+ioopBufferOffset (IOOpWrite _ _ _ bufOff _) = bufOff
+
+ioopByteCount :: IOOp m h -> ByteCount
+ioopByteCount (IOOpRead _ _ _ _ c)  = c
+ioopByteCount (IOOpWrite _ _ _ _ c) = c
```

Comment:
Not directly related to this PR, but would it make sense to turn `IOOp` into a flat record with a field `IOOpType = IOOPRead | IOOPWrite`?
Then we could just have field accessors. It also enables the `Constructed Product Result` optimisation, but that's probably not too relevant here, since the `IOOps` need to allocated for the vector either way.

Reply:
Makes sense to keep it as separate constructors!